test/framework/clusterctl/e2e_config.go

```diff
@@ -39,6 +38,11 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
+const (
+	CNIPath      = "CNI"
+	CNIResources = "CNI_RESOURCES"
```

Comment:
If we define these here, I think we will limit the CNIs to only 1 per e2e tests. 

Previously, to use a different CNI, these parameters were being changed and by setting the changed parameters to CRS as data was working fine.
CAPZ is using multiple CNIs. cc @CecileRobertMichon

Reply:
Ooh great catch @sedefsavas. That's true, we do that for IPv6 https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/test/e2e/e2e_suite_test.go#L171 (and actually have a different CNIPath defined for that and rely on the `SetCNIEnvVar` function).

Reply:
Good catch, I was not aware of the multiple CNI use case.
I have proposed a solution to solve this use case, and potentially enabling also other File transformations will be required in the future.